Scaling down logo and getting it to sit beside site title

laurelfulford opened the pull request:
Related to #345.

Updating logo styles to be similar to mockup in #119. Site title and description will float next to the logo when it's present, and the hope is that the logo will be a bit larger when the site title is hidden.

Currently this behaviour is there, but it uses 'refresh' to update 'header_textcolor' in the Customizer, which isn't great. I haven't actually reduced the image size yet, either.

Would appreciate any thoughts on improving how the Customizer handles this, if possible!

laurelfulford committed:
Building out editor styles based on current style.css. See #281.

laurelfulford committed:
Fixing formatting issue in comment in functions.php. See #281.

laurelfulford committed:
Move IE8-specific image sizing style to ie8.css, to fix image scaling weirdness in other browsers. See #305.

Commit:
Uses new get_theme_file_uri(), get_parent_theme_file_uri(), and get_parent_theme_file_path() functions, closes #176

Commit:
fixed svgs in function-icons.php

Commit:
added back-compat.php

Commit:
Merge branch 'master' into enqueue

Commit:
added space in functions.php

Commit:
adjusted spacing

davidakennedy committed:
Use new functions available for theme developers in 4.7.

* Also, add `back-compat.php` file. Twenty Seventeen can only be used in 4.7.
* Also uses `the_custom_logo` instead if custom function.
* Cleans up some spacing.

Closes #176.

See #287.

laurelfulford committed:
Work in process: scaling down logo and getting it to sit beside the site title and description. Trying to get logo to resize depending on whether the site title and description are visible. See #345.

laurelfulford committed:
Work in process: scaling down logo and getting it to sit beside the site title and description. Trying to get logo to resize depending on whether the site title and description are visible. See #345.

laurelfulford committed:
Work in process: scaling down logo and getting it to sit beside the site title and description. Trying to get logo to resize depending on whether the site title and description are visible. See #345.

laurelfulford committed:
Work in process: scaling down logo and getting it to sit beside the site title and description. Trying to get logo to resize depending on whether the site title and description are visible. See #345.

laurelfulford committed:
Fix CSS formatting based on feedback.

melchoyce commented:
How do you feel about:

> Is there some combination of CSS selectors that would let us scale down the logo if there's a title/tagline present, but let it default to its current max-width if the title/tagline are hidden?

From https://github.com/WordPress/twentyseventeen/issues/189#issuecomment-252521199?

davidakennedy committed:
Merge branch 'enqueue'

davidakennedy committed:
Fix error in custom colors call.

laurelfulford commented:
> Is there some combination of CSS selectors that would let us scale down the logo if there's a title/tagline present, but let it default to its current max-width if the title/tagline are hidden?

Geeze, I really overcomplicated this 😐  Will try that instead - thanks @melchoyce!

laurelfulford committed:
Remove 'refresh' from Customizer behavior. Reduce logo size, and update markup to fix Travis error.

laurelfulford committed:
Remove 'refresh' from Customizer behavior. Reduce logo size, and update markup to fix Travis error.

laurelfulford committed:
Remove 'refresh' from Customizer behavior. Reduce logo size, and update markup to fix Travis error.

laurelfulford committed:
Remove 'refresh' from Customizer behavior. Reduce logo size, and update markup to fix Travis error.

laurelfulford commented:
Darn it, the markup doesn't actually change whether or not the site title/description are visible - they're just hidden via CSS. 

Will try to figure out something else, but in the meantime, I've reduced the logo size.

davidakennedy committed:
Merge pull request #318 from laurelfulford/fixes/issue-281-editor-styles

Add editor styles to Twenty Seventeen.

Fixes #281.

adammacias committed:
Move Screen Reader Text Span Inside the Translation String for Edit Link

laurelfulford committed:
Merge pull request #396 from davidakennedy/enqueue2

Use new functions available for theme developers in 4.7.

davidakennedy committed:
Use `get_parent_theme_file_path` to include color patterns.

* This way, child themes can easily override the file.

Fixes #384.

celloexpressions commented:
How about toggling a body class to show/hide? That's easy to preview with `postMessage`.

laurelfulford committed:
Merge pull request #401 from davidakennedy/colorp

Use `get_parent_theme_file_path` to include color patterns.

adammacias committed:
Change the_title() to get_the_title()

laurelfulford committed:
Adding check to customizer.js, to toggle body class when the site title is shown hidden. This allows previewing the change in logo size when the site title is shown and hidden in the Customzier.

Updated styles to make logo a bit smaller on subpages.

See #345.

laurelfulford committed:
Adding check to customizer.js, to toggle body class when the site title is shown hidden. This allows previewing the change in logo size when the site title is shown and hidden in the Customzier.

Updated styles to make logo a bit smaller on subpages.

See #345.

laurelfulford commented:
> How about toggling a body class to show/hide? That's easy to preview with postMessage.

Gah, of course! Thanks @celloexpressions - I'm clearly still overcomplicating this 🙂

laurelfulford committed:
Fixing merge conflict.

laurelfulford commented:
@melchoyce I've made the logo smaller, but I could use your eyes on how this looks and acts, and on the logo's shape. Let me know if you'd like to see any changes there!

laurelfulford committed:
Adding a bit more logic, so if the custom header is removed from the front page, the smaller logo size is used.

Giving the logo flex width rather than flex height.

Adding mobile styles to make sure logo isn't too large.

See #189.

laurelfulford committed:
Adding a bit more logic, so if the custom header is removed from the front page, the smaller logo size is used. See #189.

celloexpressions commented:
In the add theme support, I'm guessing `flex-width` should be true, and possibly not the height. That way a larger logo with a title in the image can take over the space of the site title and tagline when they're hidden. Since we're setting the max height in CSS as up to 200px, that should be the `height`, with the width being whatever (probably 200 to encourage a square aspect ratio) and flex-width, not height, being true.

laurelfulford committed:
Merge pull request #399 from adammacias/screen-reader-text

Move Screen Reader Text Span Inside the Translation String for Edit Link

laurelfulford committed:
Update CONTRIBUTORS.md

melchoyce committed:
Adding styles to audio and video playlists

melchoyce committed:
Better selectors and em padding, courtesy of @davidakennedy

melchoyce committed:
Split color vs. non-color styles, and prefix light colors with the light class

davidakennedy committed:
Merge pull request #407 from melchoyce/update/playlists

Add basic playlist styling for audio and video.

Fixes #137.

davidakennedy committed:
Merge pull request #338 from laurelfulford/fixes/issue-305-thumbnails-scaled-up

Move IE8-specific image sizing style to `ie8.css`.

Fixes #305.

laurelfulford committed:
Resolving conflicts.

laurelfulford committed:
Removing larger logo image widths based on feedback. See #189.

laurelfulford commented:
I turned this PR into a bit of a mess, so have replaced with #434. Changes actually related to the logo have all made their way over there 🙂
